src/Nix/Effects/Basic.hs

```diff
@@ -235,13 +225,13 @@ findPathM = findPathBy existingPath
     pure $ if exists then Just apath else Nothing
 
 defaultImportPath
-  :: (MonadNix e t f m, MonadState (HashMap FilePath NExprLoc) m)
+  :: (MonadNix e t f m, MonadState (HashMap FilePath NExprLoc, b) m)
```

Comment:
We could move to a more involved type. Possibly something like `Has ExprCache` and `Has DrvHashCache` but for StateT instead of ReaderT. Looks a lot lensy to me. I did not want to bother generalizing too much because

1. This might move to hnix-store if we end-up weaving the effect stacks
2. There does not seem to be a need for more stateful values
3. Usage of this state is limited to a few fucntions.

Reply:
Oh sure, I was just musing :)